### PR TITLE
Return all matching keys within a directory for a given resource pattern

### PR DESCRIPTION
**Pull request by clorenz**

### CODE REVIEW

File: dc-commons-file/src/main/java/de/digitalcollections/commons/file/backend/impl/FileResourceRepositoryImpl.java [hunk 1]
```diff
@@ -158,7 +173,9 @@ public FileResource find(String key, FileResourcePersistenceType resourcePersist
   public InputStream getInputStream(URI resourceUri) throws ResourceIOException {
     try {
       String location = resourceUri.toString();
-      LOGGER.info("Getting inputstream for location '{}'.", location);
+      if (LOGGER.isDebugEnabled()) {
```
Comment (datazuul) [inline dc-commons-file/src/main/java/de/digitalcollections/commons/file/backend/impl/FileResourceRepositoryImpl.java]: guard not really necessary as no costly operation in logging statement, see https://stackoverflow.com/a/12953090

File: dc-commons-file/src/main/java/de/digitalcollections/commons/file/backend/impl/FileResourceRepositoryImpl.java [hunk 2]
```diff
@@ -240,7 +257,9 @@ public void write(FileResource resource, InputStream payload) throws ResourceIOE
       }
 
       Files.createDirectories(Paths.get(uri).getParent());
-      LOGGER.info("Writing: " + uri);
+      if (LOGGER.isDebugEnabled()) {
```
Comment (datazuul) [inline dc-commons-file/src/main/java/de/digitalcollections/commons/file/backend/impl/FileResourceRepositoryImpl.java]: see above comment

File: dc-commons-file/src/main/java/de/digitalcollections/commons/file/backend/impl/FileResourceRepositoryImpl.java [hunk 3]
```diff
@@ -260,4 +279,79 @@ public void write(FileResource resource, String input) throws ResourceIOExceptio
     }
   }
 
+  @Override
+  public Set<String> findKeys(String keyPattern, FileResourcePersistenceType resourcePersistenceType) throws ResourceIOException {
+    // The pattern for valid keys is the original pattern without any brackets inside, but surrounded with one bracket.
+    Pattern validKeysPattern = Pattern.compile(
+        "("
+            + keyPattern.replace("(","").replace(")","").replace("^","").replace("$","")
```
Comment (datazuul) [inline dc-commons-file/src/main/java/de/digitalcollections/commons/file/backend/impl/FileResourceRepositoryImpl.java]: what if more open brackets, better use replaceAll?

File: dc-commons-file/src/main/java/de/digitalcollections/commons/file/backend/impl/FileResourceRepositoryImpl.java [hunk 4]
```diff
@@ -260,4 +279,79 @@ public void write(FileResource resource, String input) throws ResourceIOExceptio
...
+            + ")");
+
+    if (resourcePersistenceType != FileResourcePersistenceType.RESOLVED) {
```
Comment (datazuul) [inline dc-commons-file/src/main/java/de/digitalcollections/commons/file/backend/impl/FileResourceRepositoryImpl.java]: move before validKeyPatterns operation

File: dc-commons-file/src/test/java/de/digitalcollections/commons/file/backend/impl/FileResourceRepositoryImplTest.java [hunk 5]
```diff
@@ -20,11 +27,18 @@
...
 import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
 import org.w3c.dom.Document;
 import org.w3c.dom.Node;
 
+import static de.digitalcollections.model.api.identifiable.resource.enums.FileResourcePersistenceType.RESOLVED;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+//@Ignore
```
Comment (datazuul) [inline dc-commons-file/src/test/java/de/digitalcollections/commons/file/backend/impl/FileResourceRepositoryImplTest.java]: remove comment

File: dc-commons-file/src/main/java/de/digitalcollections/commons/file/backend/impl/FileResourceRepositoryImpl.java (hunk 3 above)
Comment (clorenz) [reply dc-commons-file/src/main/java/de/digitalcollections/commons/file/backend/impl/FileResourceRepositoryImpl.java]: String.replace replaces all occurrences of the term; String.replaceAll does the same, but with aregular expression

**Pull request merged**